test/test_intensify/test_parallel_successive_halving.py

```diff
@@ -285,6 +289,65 @@ def test_sh_scheduling(self):
         self.PSH.last_active_instance = 2
         self.assertCountEqual(list(self.PSH._sh_scheduling(strategy='serial')), [3, 0, 1, 2])
 
+    def test_sh_scheduling_more_advanced_iteration(self):
+        """Ensures that we prioritize the more advanced stage iteration"""
+
+        def add_sh_mock(stage, config_inst_pairs):
+            sh = mock.Mock()
+            sh.run_tracker = []
+            for i in range(config_inst_pairs):
+                sh.run_tracker.append((i, i, i))
+            sh.stage = stage
+            return sh
+
+        # Add more SH to make testing interesting
+        self.tae.num_workers.return_value = 5
+        self.PSH.sh_instances[0] = add_sh_mock(stage=1, config_inst_pairs=6)
+        self.PSH.sh_instances[1] = add_sh_mock(stage=1, config_inst_pairs=2)
+
+        # We only have two configurations in the same stage.
+        # In this case, we want to prioritize the one with more launched runs
+        # that is zero
+        self.assertCountEqual(
+            list(self.PSH._sh_scheduling(strategy='more_advanced_iteration')),
+            [0, 1]
+        )
+
+        # One more instance comparison to be supper safe
+        self.PSH.sh_instances[2] = add_sh_mock(stage=1, config_inst_pairs=7)
+        self.assertCountEqual(
+            list(self.PSH._sh_scheduling(strategy='more_advanced_iteration')),
+            [2, 0, 1]
+        )
+
+        # Not let us add a more advanced stage run
+        self.PSH.sh_instances[3] = add_sh_mock(stage=2, config_inst_pairs=1)
+        self.assertCountEqual(
+            list(self.PSH._sh_scheduling(strategy='more_advanced_iteration')),
+            [3, 2, 0, 1]
+        )
+
+        # Make 1 the oldest stage
+        self.PSH.sh_instances[1] = add_sh_mock(stage=4, config_inst_pairs=1)
+        self.assertCountEqual(
+            list(self.PSH._sh_scheduling(strategy='more_advanced_iteration')),
+            [1, 3, 2, 0]
+        )
+
+        # Add a new run that's empty
+        self.PSH.sh_instances[4] = add_sh_mock(stage=0, config_inst_pairs=0)
+        self.assertCountEqual(
+            list(self.PSH._sh_scheduling(strategy='more_advanced_iteration')),
+            [1, 3, 2, 0, 4]
+        )
+
+        # Lastly make 0 stage 4 but with not as many instances as 1
```

Comment:
I believe there's something wrong here - the comment says that we change sh_instance 0, but the code changes sh_instance 4, yet the assert checks that sh_instance 0 was changed.

Reply:
Sorry, I used assert count equal but according to the documentation:
```
a and b have the same elements in the same number, regardless of their order.
```

So I am gonna move to assert equal